test/src/androidTest/kotlin/io/realm/shared/RealmTests.kt

```diff
@@ -126,4 +137,215 @@ class RealmTests {
             otherRealm.close()
         }
     }
+
+    @Suppress("invisible_member")
+    @Test
+    fun write() = runBlocking {
+        val name = "Realm"
+        val child: Child = realm.write {
+            this.copyToRealm(Child()).apply { this.name = name }
+        }
+        assertEquals(name, child.name)
+        val objects = realm.objects<Child>()
+        val childFromResult = objects[0]
+        assertEquals(name, childFromResult.name)
+    }
+
+    @Suppress("invisible_member")
+    @Test
+    fun exceptionInWriteWillRollback() = runBlocking {
+        class CustomException : Exception()
+
+        assertFailsWith<CustomException> {
+            realm.write {
+                val name = "Realm"
+                this.copyToRealm(Child()).apply { this.name = name }
+                throw CustomException()
+            }
+        }
+        assertEquals(0, realm.objects<Child>().size)
+    }
+
+    @Test
+    fun writeBlocking() {
+        val managedChild = realm.writeBlocking { copyToRealm(Child().apply { name = "John" }) }
+        assertTrue(managedChild.isManaged())
+        assertEquals("John", managedChild.name)
+    }
+
+    @Suppress("invisible_member")
+    @Test
+    fun writeBlockingAfterWrite() = runBlocking {
+        val name = "Realm"
+        val child: Child = realm.write {
+            this.copyToRealm(Child()).apply { this.name = name }
+        }
+        assertEquals(name, child.name)
+        assertEquals(1, realm.objects<Child>().size)
+
+        realm.writeBlocking {
+            this.copyToRealm(Child()).apply { this.name = name }
+        }
+        Unit
+    }
+
+    @Suppress("invisible_member")
+    @Test
+    fun exceptionInWriteBlockingWillRollback() {
+        class CustomException : Exception()
+        assertFailsWith<CustomException> {
+            realm.writeBlocking {
+                val name = "Realm"
+                this.copyToRealm(Child()).apply { this.name = name }
+                throw CustomException()
+            }
+        }
+        assertEquals(0, realm.objects<Child>().size)
+    }
+
+    @Test
+    @Suppress("invisible_member")
+    fun simultaneousWritesAreSerialized() = runBlocking {
+        val jobs: List<Job> = IntRange(0, 9).map {
+            launch {
+                realm.write {
+                    copyToRealm(Parent())
+                }
+            }
+        }
+        jobs.map { it.join() }
+
+        // Ensure that all writes are actually committed
+        realm.close()
+        assertTrue(realm.isClosed())
+        realm = Realm.open(configuration)
+        assertEquals(10, realm.objects(Parent::class).size)
+    }
+
+    @Test
+    @Suppress("invisible_member")
+    fun writeBlockingWhileWritingIsSerialized() = runBlocking {
+        val mutex = Mutex(true)
+        val exit = Mutex(true)
+        val write = async {
+            realm.write {
+                mutex.unlock()
+                PlatformUtils.sleep(1.milliseconds)
```

Comment:
Why not using `delay`?

Reply:
Delay is a suspending operation and the write block is currently not suspending as this could lead to deadlocks.

Reply:
Delay is a suspending operation and the write block is currently not suspending as this could lead to deadlocks.